### DISCUSSION TITLE
Question: do you autogenerate your readme Tables of Contents?

### DISCUSSION THREAD

**Discussion by ElectricRCAircraftGuy** [mmistakes/so-simple-theme#398] [Q&A] [answered]
You have really nice tables of contents in your readmes--ex: https://github.com/mmistakes/so-simple-theme#table-of-contents. Do you autogenerate these somehow?--(which would be ideal), or do you have to manually create these?

If you autogenerate them, I'd like to know how. 

Thanks! 

EXCELLENT Jekyll themes by the way! Thanks so much for contributing them to the community.

**Comment by mmistakes** [comment, marked as answer]
They're autogenerated. I use this VS Code extension, which takes all the headines in my `README` file and converts into a TOC.

https://github.com/AlanWalk/Markdown-TOC

**Comment by ElectricRCAircraftGuy** [comment]
Thank you! Because of your response, I discovered this amazing [MarkdownTOC plugin](https://github.com/naokazuterada/MarkdownTOC) for Sublime Text 3, and wrote about it in my answer here: https://stackoverflow.com/questions/11948245/markdown-to-create-pages-and-table-of-contents/64656967#64656967.
